### DISCUSSION TITLE
Selection manipulation minor mode?

### DISCUSSION THREAD

**Discussion by sicher** [helix-editor/helix#8731] [General]
Hey, maybe I am the only one but I have a really hard time learning which of these powerful and very useful selection commands is which:

- flip_selections (A-;)
- collapse_selection (;)
- keep_primary_selection (,)
- remove_primary_selection (A-,)
- ...and a few more...

I wonder if, in addition to the current keybindings, we could add a "selections" minor mode with these and other selection commands. I have one in my local config with mnemonic bindings (f - flip_selections, c - collapse etc) and wonder if you think this is something that would be a good as a default?

**Comment by djpowers** [comment]
I'd love to see something of the sort as well. There's already [Select / extend mode](https://docs.helix-editor.com/keymap.html#select--extend-mode) that's entered with the `v` key, so hopefully the [Selection manipulation](https://docs.helix-editor.com/keymap.html#selection-manipulation) commands could be namespaced under a "Selection" minor mode and avoid confusion there.

This would also have the benefit of providing an alternative to the `Alt-*` keymaps, which I've seen people looking for:
- #6853
- [Reddit: Anyone hates the Alt key?](https://www.reddit.com/r/HelixEditor/comments/1730bm5/anyone_hates_the_alt_key/)
- [Reddit: Remapping alt key](https://www.reddit.com/r/HelixEditor/comments/1bvlggg/remapping_alt_key/)

Since most of the keymaps using `alt` are in the selection manipulation category, such a dedicated mode would allow users a way to avoid using that key, while also improving discoverability (as the original post notes).

**Reply by sicher** [reply]
I have been thinking about something like the following instead of a new "selection" minor mode:

1. Repurpose the 'm' minor mode. Instead of the pretty small set of 'match' commands, it could instead be renamed 'modify' or 'manipulate'. Then it could house a good set of selection manipulations, as well as the current surround and select.
2. For consistency, move "goto matching bracket" from 'm' to the 'g' minor mode. Not sure what a good mnemonic would be for it though, since both 'm' and 'b' is already in there.
